gemini/use-cases/applying-llms-to-data/docai-gemini/cf-trigger-event/src/extractor.py

```diff
@@ -0,0 +1,147 @@
+        gcs_temp_uri: str,
+        processor_version_id: str,
+        timeout: int = 400,
+    ):
+        super().__init__(project_id, location, processor_id, processor_version_id)
+        self.gcs_output_uri = gcs_output_uri
+        self.timeout = timeout
+        self.storage_client = storage.Client()
+        self.temp_file_uploader = TempFileUploader(gcs_temp_uri)
+
+    def process_document(self, file_path: str, mime_type: str) -> documentai.Document:
+        gcs_input_uri = self.temp_file_uploader.upload_file(file_path)
+        document = self._process_document_batch(gcs_input_uri, mime_type)
+        self.temp_file_uploader.delete_file()
+        return document
+
+    def _process_document_batch(
+        self, gcs_input_uri: str, mime_type: str
+    ) -> documentai.Document:
+        gcs_document = documentai.GcsDocument(
+            gcs_uri=gcs_input_uri, mime_type=mime_type
+        )
+        gcs_documents = documentai.GcsDocuments(documents=[gcs_document])
+        input_config = documentai.BatchDocumentsInputConfig(gcs_documents=gcs_documents)
+
+        gcs_output_config = documentai.DocumentOutputConfig.GcsOutputConfig(
+            gcs_uri=self.gcs_output_uri
+        )
+        output_config = documentai.DocumentOutputConfig(
+            gcs_output_config=gcs_output_config
+        )
+
+        request = documentai.BatchProcessRequest(
+            name=self.processor_name,
+            input_documents=input_config,
+            document_output_config=output_config,
+        )
+
+        operation = self.client.batch_process_documents(request)
+        try:
+            print(f"Waiting for operation ({operation.operation.name}) to complete...")
+            operation.result(timeout=self.timeout)
+        except (RetryError, InternalServerError) as e:
+            print(e.message)
+
+        metadata = documentai.BatchProcessMetadata(operation.metadata)
+        if metadata.state != documentai.BatchProcessMetadata.State.SUCCEEDED:
+            raise ValueError(f"Batch Process Failed: {metadata.state_message}")
+
+        # Retrieve the processed document from GCS
+        for process in list(metadata.individual_process_statuses):
+            matches = re.match(r"gs://(.*?)/(.*)", process.output_gcs_destination)
+            if not matches:
+                print(
+                    "Could not parse output GCS destination:",
+                    process.output_gcs_destination,
+                )
+                continue
+
+            output_bucket, output_prefix = matches.groups()
+            output_blobs = self.storage_client.list_blobs(
+                output_bucket, prefix=output_prefix
+            )
+            for blob in output_blobs:
+                if blob.content_type == "application/json":
+                    print(f"Fetching {blob.name}")
+                    return documentai.Document.from_json(
+                        blob.download_as_bytes(), ignore_unknown_fields=True
```

Comment:
I'd recommend using Document AI Toolbox for this.

https://cloud.google.com/document-ai/docs/toolbox